.github/workflows/test-full-sync.yml

```diff
@@ -58,6 +58,41 @@ jobs:
         with:
           short-length: 7
 
+      - name: Authenticate to Google Cloud
+        id: auth
+        uses: google-github-actions/auth@v0.7.0
+        with:
+          workload_identity_provider: 'projects/143793276228/locations/global/workloadIdentityPools/github-actions/providers/github-oidc'
+          service_account: 'github-service-account@zealous-zebra.iam.gserviceaccount.com'
+          token_format: 'access_token'
+
+      # If we have accidentally changed the format, but not changed the version,
+      # we want to run with the old cached state, so other jobs fails.
+      - name: Get specific changed files
+        id: changed-files-specific
+        uses: tj-actions/changed-files@v18.6
+        with:
+          files: |
+            zebra-state/**/constants.rs
+
+      # Before executing any further steps, validate the local state and remote version are the same,
+      # or at least that the local state version is lower than the available cached state version.
+      #
+      # This step should only be validated if there are no changes in the files specifified
+      # in the changed-files-specific step.
+      - name: Validate constants.rs version vs cached state version
+        id: validate-state-version
+        if: ${{ steps.changed-files-specific.outputs.any_changed != 'true' }}
```

Comment:
Question: do we also want to run this check when the workflow file changes?

Reply:
Fixed by: https://github.com/ZcashFoundation/zebra/pull/4074/commits/8f24f0f0dc62c307adc43d74e283e27950f9abe4